nexus/src/app/sagas/instance_delete.rs

```diff
@@ -68,6 +75,82 @@ impl NexusSaga for SagaInstanceDelete {
 
 // instance delete saga: action implementations
 
+async fn sid_delete_network_config(
+    sagactx: NexusActionContext,
+) -> Result<(), ActionError> {
+    let params = sagactx.saga_params::<Params>()?;
+    let opctx = OpContext::for_saga_action(&sagactx, &params.serialized_authn);
+    let osagactx = sagactx.user_data();
+    let dpd_client = &osagactx.nexus().dpd_client;
+    let datastore = &osagactx.datastore();
+    let log = sagactx.user_data().log();
+
+    debug!(log, "fetching external ip addresses");
+
+    let external_ips = &datastore
+        .instance_lookup_external_ips(&opctx, params.authz_instance.id())
+        .await
+        .map_err(ActionError::action_failed)?;
+
+    let mut entries_to_remove: Vec<ExternalIp> = vec![];
+
+    for entry in external_ips {
```

Comment:
Same thing here about idempotency. We need a different saga action for each external IP I think.

Reply:
Persisting convo from chat:

It doesn't seem that any of the other saga nodes in `instance_delete.rs` are using subsagas, so here we're going to ensure that we have attempted to nat entries for all external ips before checking errors.